mlx_graphs/datasets/elliptic.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional
+
+import mlx.core as mx
+import pandas as pd
+
+from mlx_graphs.data import GraphData
+from mlx_graphs.datasets.dataset import Dataset
+from mlx_graphs.datasets.utils import download, extract_archive
+
+
+class EllipticBitcoinDataset(Dataset):
+    """The Elliptic Bitcoin dataset of Bitcoin transactions from the
+    `"Anti-Money Laundering in Bitcoin: Experimenting with Graph Convolutional
+    Networks for Financial Forensics" <https://arxiv.org/abs/1908.02591>`_
+    paper.
+
+    :class:`EllipticBitcoinDataset` maps Bitcoin transactions to real entities
+    belonging to licit categories (exchanges, wallet providers, miners,
+    licit services, etc.) versus illicit ones (scams, malware, terrorist
+    organizations, ransomware, Ponzi schemes, etc.)
+
+    There exists 203,769 node transactions and 234,355 directed edge payments
+    flows, with two percent of nodes (4,545) labelled as illicit, and
+    twenty-one percent of nodes (42,019) labelled as licit.
+    The remaining transactions are unknown
+
+    Args:
+       base_dir: Directory where to store dataset files. Default is
+            in the local directory ``.mlx_graphs_data/``.
+
+    Example:
+
+    .. code-block:: python
+
+        from mlx_graphs.datasets import EllipticBitcoinDataset
+
+        dataset = EllipticBitcoinDataset()
+        >>> dataset.num_graphs
+        1
+
+        >>> dataset[0]
+        GraphData(
+            edge_index(shape=(2, 234355), int64)
+            node_features(shape=(203769, 165), float32)
+            node_labels(shape=(203769,), int64)
+            train_mask(shape=(203769,), bool)
+            test_mask(shape=(203769,), bool))
+    """
+
+    def __init__(self, base_dir: Optional[str] = None):
+        super().__init__(name="ellipticBitcoin", base_dir=base_dir)
+
+    @property
+    def raw_file_names(self):
+        return [
+            "elliptic_txs_features.csv",
+            "elliptic_txs_edgelist.csv",
+            "elliptic_txs_classes.csv",
+        ]
+
+    def download(self):
+        # This url is unable to download the data for elliptic bitcoin dataset
+        # lets try with pytorch geometric to verify the data
+        url = "https://data.pyg.org/datasets/elliptic/"
+        for files in self.raw_file_names:
+            download(f"{url}{files}.zip", self.raw_path)
+            extract_archive(f"{self.raw_path}/{files}.zip", self.processed_path)
```

Comment:
why are you extracting the data in `processed_path`? I guess it should go in `raw_path`

Reply:
Updated in the new commit. I have also added pandas in the dependency section since the tests were failing